packages/composable-controller/src/ComposableController.ts

```diff
@@ -16,8 +15,18 @@ import type { Patch } from 'immer';
 
 export const controllerName = 'ComposableController';
 
-export const INVALID_CONTROLLER_ERROR =
-  'Invalid controller: controller must have a `messagingSystem` or be a class inheriting from `BaseControllerV1`.';
+// TODO: Replace with type guard once superclass for messageable non-controllers has been defined.
```

Comment:
If the goal is to figure out which controllers do not have state, can we check for the absence of the `state` property?

Reply:
Unfortunately only `AssetsContractController` lacks a `state` property.

`NftDetectionController` and `TokenDetectionController` both have a `state` property which is assigned an empty object of type `Record<never, never>`, because both inherit from `BaseController`.

These inconsistencies make the changes in this PR less type-safe than they could be. This ties in to one of the motivations for the [MessengerClient/Messageable ADR](https://github.com/MetaMask/decisions/pull/41). which is to align inheritance for stateless non-controllers.

Reply:
Oh, I see. My thought is that we could clean this up by assuming that the non-controllers do not inherit from BaseController. I forgot that we will do that through the MessengerClient/Messageable ADR though.

Reply:
There's currently no common supertype for the stateless non-controllers that is narrower than an object with a 'name' property. 

Ideally, we would be able to exclude all classes with properties 'name', 'messagingSystem' and without 'state', 'metadata', but since that's currently not an option, I went for just explicitly specifying the non-controllers, so we wouldn't have to deal with edge cases in the input.

I refactored the logic a bit. Hopefully this makes the intention a bit clearer: https://github.com/MetaMask/core/pull/4943/commits/85389bbb7e55ef77cf48c0fd1e8f4a17d2cd0323